### ISSUE TITLE
Out of workspace files: Changes from git operations not reflected

### ISSUE THREAD

**Issue by constantincerdan** [Microsoft/vscode#2592]
As above. I've been testing it out by switching branches in git using terminal and the file does not update. Sublime & Atom do it immediately, but with Code you have to switch out and back into the file in order for it to update.

**Comment by bpasero** [comment]
@constantincerdan on which OS is this? does it work when you edit one of the working files with another editor outside of Code?

I assume the working files are not part of the folder you opened? Or are you not opening a folder at all?

**Comment by constantincerdan** [comment]
It's on Mac OSX. It doesn't work at all, even when the files are edited using another editor. I haven't opened any folders and have instead opened the file independently. 

-----Original Message-----

**Comment by bpasero** [comment]
Weird, works for me. See my recording and maybe see what I need to do different to reproduce:

![untitled](https://cloud.githubusercontent.com/assets/900690/12718163/e8cadb38-c8ec-11e5-973e-c4457162a7b0.gif)

**Comment by constantincerdan** [comment]
I just tested it now by changing it in a different editor, and this time it did auto update. However, it still doesn't work when switching branches in git (using Terminal). See my recording below:

http://g.recordit.co/HYlpJq5ZSB.gif

**Comment by bpasero** [comment]
I can reproduce and it looks like a bug in node.js: https://github.com/nodejs/node/issues/5039

**Comment by harangue** [comment]
@bpasero I guess [this is intended behavior](https://github.com/nodejs/node/pull/6099). Maybe https://github.com/paulmillr/chokidar would work better?

**Comment by harangue** [comment]
Apparently this is [intended behavior](https://github.com/nodejs/node/pull/6099). Maybe chokadir can figure out a fix? I kind of doubt it though...

**Comment by bpasero** [comment]
https://github.com/Microsoft/vscode/issues/15111

**Comment by fatonsopa** [comment]
Hi guys, 

this is still happening:

npm -v 3.10.10
node -v v6.11.5
MacOS Sierra 10.12.6

The file.watch is triggering the change only once per file. In windows is working perfectly.

Researched a lot but couldn't find a working solution. Any idea?

Thanks,
Faton
